http_handler.go

```diff
@@ -43,23 +46,35 @@ func Trace(handler http.Handler) http.Handler {
 // necessary.
 
 // In addition to tracing capabilities this function also provides graceful
-// shutdown of the server. When a SIGTERM signal is received by the server
-// it will stop accepting new connections but it will keep it running until
-// all the existing connections have been closed. It will kill the server
+// shutdown of the server. When a SIGTERM or SIGINT signal is received by the
+// server it will stop accepting new connections but it will keep it running
+// until all the existing connections have been closed. It will kill the server
 // after a default timeout of 25 seconds if there are still any pending
 // connections.
 
-func ListenAndServe(addr string, handler http.Handler) error {
+func ListenAndServe(addr string, handler http.Handler) {
```

Comment:
The reason this function was named ListenAndServe in the first place was to make it drop-in-compatible with the stdlib http.ListenAndServe, which is wasn't quite even before this change, since it did its own style of logging. Now it is even more different, so the API seems like a huge hack to me. I understand the desire to do it this way, to reduce the burden of porting all services to have graceful shutdown, but I also think it is very hacky.  In my opinion we should have much higher standards in shared code.